### PR TITLE
[TEST][ICEBERG][DELTA] Introduce row level operation test for data lake format

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Introduce row level operation test for data lake format and remove redundant tests

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaMetadataTests.scala [hunk 1]
```diff
@@ -87,162 +87,4 @@ trait DeltaMetadataTests extends HiveJDBCTestHelper with DeltaSuiteMixin {
       assert(!rs3.next())
     }
   }
-
-  test("get type info") {
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaMetadataTests.scala:91]: It's independent with table format, but engine.

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/IcebergRowLevelOperationTests.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.operation
+
+import org.apache.kyuubi.IcebergSuiteMixin
+
+trait IcebergRowLevelOperationTests extends HiveJDBCTestHelper with IcebergSuiteMixin {
+
+  private def createAndInitTable(tableName: String)(records: Seq[(Int, String)]): Unit = {
```
Comment (yaooqinn) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/IcebergRowLevelOperationTests.scala:24]: we can have a statement as parameter here

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/SparkMetadataTests.scala [hunk 3]
```diff
@@ -287,4 +292,188 @@ trait SparkMetadataTests extends HiveJDBCTestHelper {
...
+        () => metaData.autoCommitFailureClosesAllResultSets,
+        () => metaData.getClientInfoProperties,
+        () => metaData.getFunctionColumns("", "%", "%", "%"),
+        () => metaData.getPseudoColumns("", "%", "%", "%"),
+        () => metaData.generatedKeyAlwaysReturned).foreach { func =>
+        val e = intercept[SQLFeatureNotSupportedException](func())
+        assert(e.getMessage === "Method not supported")
+      }
+
+      import org.apache.kyuubi.KYUUBI_VERSION
```
Comment (yaooqinn) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/SparkMetadataTests.scala:407]: let's fix this too

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.operation
+
+import org.apache.kyuubi.DeltaSuiteMixin
+
+trait DeltaRowLevelOperationTests extends HiveJDBCTestHelper with DeltaSuiteMixin {
```
Comment (yaooqinn) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala:22]: how about we rename it to DML and bring more cases in?

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/IcebergRowLevelOperationTests.scala (hunk 2 above)
Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/IcebergRowLevelOperationTests.scala:24]: Do you mean
```
def createAndInitTable(tableName: String)
                      (records: Seq[(Int, String)])
                      (initSql: Option[String] = None): Unit
```

Comment (yaooqinn) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/IcebergRowLevelOperationTests.scala:24]: private def createAndInitTable(statement: Statement, tableName: String)

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala (hunk 4 above)
Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala:22]: Row level operation is the spark offical term introduced in https://issues.apache.org/jira/browse/SPARK-35801, I'm planning to add other cases later in `TimeTravelTests` `SecondaryIndexTests` etc to reflect different capbilities in different table format

Comment (yaooqinn) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala:22]: The complexity of mixins brings more overhead of maintenance. I have never experienced using traits for classification, not abstraction

Comment (ulysses-you) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala:22]: what's the behavior of iceberg or delta ?

Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala:22]: Any suggestion for classification expect traits in scalatest? @yaooqinn

Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala:22]: > what's the behavior of iceberg or delta ?

I don't get your point.

Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/operation/DeltaRowLevelOperationTests.scala:22]: BTW, I explained the idea in #1287, if we reject to use traits for classification, we need to reconsider the test classes hierarchical

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**